### DISCUSSION TITLE
502 bad gateway using docker

### DISCUSSION THREAD

**Discussion by halindraprakoso** [Budibase/budibase#5908] [Help] [answered]
Hello,

I am having problem running budibase using docker on m1 mac. I added `platform: linux/amd64` under couchdb-service in the docker-compose.yaml but still getting 502 bad gateway.

**Comment by melohagan** [comment, marked as answer]
Hi @halindraprakoso 

Are you using the latest docker-compose file? https://raw.githubusercontent.com/Budibase/budibase/master/hosting/docker-compose.yaml

The installation may simply have failed. Try stopping and removing the containers and doing another 
`docker-compose up`

Then have a look and make sure all of the containers are running successfully:
<img width="454" alt="Screenshot 2022-05-16 at 17 15 10" src="https://user-images.githubusercontent.com/101575380/168637649-60e922be-74e8-4fcb-ab7d-4fc19e8ce963.png">
*I have a few extra db services, don't worry about them*.

Or `docker ps -a` if you don't have docker desktop.

**Reply by halindraprakoso** [reply]
Hi @melohagan! yes I tried that and the one generated by the cli and still the same bad gateway error even though the containers are running. 
The app seems to be crashing for some reason
```js
bbapps exited with code 137
bbapps                         | yarn run v1.22.17
bbapps                         | $ node dist/index.js
bbapps                         | Killed
bbapps                         | error Command failed with exit code 137.
bbapps                         | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
bbapps exited with code 137
bbapps                         | yarn run v1.22.17
bbapps                         | $ node dist/index.js
bbapps                         | Killed
bbapps                         | error Command failed with exit code 137.
bbapps                         | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
bbapps exited with code 137


budibase-couchdb-service-1     | [main] INFO clouseau.main - Clouseau running as clouseau@127.0.0.1
bbworker                       | Worker running on {"address":"::","family":"IPv6","port":4003}
bbworker                       | error Command failed with signal "SIGKILL".
bbworker                       | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
bbworker exited with code 1
bbworker                       | yarn run v1.22.17
bbworker                       | $ node dist/index.js
budibase-couchdb-service-1     | [main] INFO clouseau.main - Clouseau running as clouseau@127.0.0.1
bbapps                         | Killed
bbapps                         | error Command failed with exit code 137.
bbapps                         | info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

There was also a crash report on couch_prometheus_server

**Reply by melohagan** [reply]
>Exist code 137 means the build process ran out of memory. You will need to increase the amount of memory available or enable swap.

Try allocating more memory and CPU to your docker setup. This is my setup - although this should be much more than needed. 

<img width="454" alt="Screenshot 2022-05-16 at 19 54 45" src="https://user-images.githubusercontent.com/101575380/168662574-a971a86b-c49f-4591-b021-c5322b97d170.png">

**Reply by halindraprakoso** [reply]
oh dang it's so simple. thanks! it's working now
